src/components/dropdownSearch/DropdownSearchReducers.tsx

```diff
@@ -111,11 +154,17 @@ export const dropdownSearchReducer = (state: IDropdownSearchState = dropdownSear
         return {
           ...state,
           id: action.payload.id,
-          selectedOption: state.activeOption,
           isOpened: false,
+          selectedOptions: state.selectedOptions.push(state.activeOption),
```

Comment:
and if you have objects in your queue...those too will remain references and not new objects

That being said, I don't know how much side effects that would cause in your particular case.